### PR TITLE
Poly perch command

### PR DESCRIPTION
**Pull request by tf-4**
## About The Pull Request

Port of https://github.com/Skyrat-SS13/Skyrat13/pull/3291 - credit @NotRanged for the original code, modified slightly for this PR.

image

## How This Contributes To The Skyrat Roleplay Experience

Cute flavour

## Changelog

:cl:
add: Poly now listens to basic voice commands, but she only recognizes the CE's authority! Say "Poly, up" or "Poly, perch" to get her on your shoulder, or "Poly, off" or "Poly, down" to get her off again!
/:cl:

### CODE REVIEW

**Review by GoldenAlpharex — APPROVED** [review]
Code looks good, a few magic numbers but it was mostly not your code, it was modularized upstream code, so I won't make a big fuss of it.  If you'd like to fix these, though, it would be nice.

File: modular_skyrat/modules/poly_commands/parrot.dm [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+/mob/living/simple_animal/parrot
+	/// Whether the parrot is on a human's shoulder or not
+	var/buckled_to_human = FALSE
+
+/mob/living/simple_animal/parrot/Hear(message, atom/movable/speaker, message_langs, raw_message, radio_freq, list/spans, list/message_mods = list())
+	. = ..()
+	if(check_command(message, speaker))
+		return
+	to_chat(speaker, span_warning("Command not recognised!"))
```
Comment (tf-4) [inline modular_skyrat/modules/poly_commands/parrot.dm]:
```suggestion
```
debug code bad

**Comment by tf-4** [comment]
> Code looks good, a few magic numbers but it was mostly not your code, it was modularized upstream code, so I won't make a big fuss of it. If you'd like to fix these, though, it would be nice.

I might do that upstream then mirror it to this when that goes through tbh

**Comment by GoldenAlpharex** [comment]
> > Code looks good, a few magic numbers but it was mostly not your code, it was modularized upstream code, so I won't make a big fuss of it. If you'd like to fix these, though, it would be nice.
> 
> I might do that upstream then mirror it to this when that goes through tbh

Want this to stay open until then?

**Comment by tf-4** [comment]
> > > Code looks good, a few magic numbers but it was mostly not your code, it was modularized upstream code, so I won't make a big fuss of it. If you'd like to fix these, though, it would be nice.
> > 
> > 
> > I might do that upstream then mirror it to this when that goes through tbh
> 
> Want this to stay open until then?

Nah I'll just address it when it happens, I don't know when I'll get around to doing that upstream and don't want this to sit here for ages

**Comment by GoldenAlpharex** [comment]
Yeet

**Pull request merged**